presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPartition.java

```diff
@@ -14,44 +14,57 @@
 
 package com.facebook.presto.hudi;
 
+import com.facebook.presto.common.predicate.NullableValue;
 import com.facebook.presto.hive.metastore.Storage;
+import com.facebook.presto.spi.ColumnHandle;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
+import static com.google.common.base.MoreObjects.toStringHelper;
 import static java.util.Objects.requireNonNull;
 
 public class HudiPartition
 {
-    private final String name;
+    private final String tableName;
+    private final String partitionName;
     private final List<String> values;
     private final Map<String, String> keyValues;
     // TODO: storage and dataColumns is required from MOR record cursor, might be able to remove later
     private final Storage storage;
     private final List<HudiColumnHandle> dataColumns;
+    private Map<ColumnHandle, NullableValue> keys;
```

Comment:
We prefer to use immutable objects in Presto. Let's keep `keys` final and add a method like `withXXX` to create a new HudiPartition instance with new keys value.